cohort-api/src/main/kotlin/com/sksamuel/cohort/HealthCheckRegistry.kt

```diff
@@ -167,11 +162,14 @@ class HealthCheckRegistry(
 
       scheduler.scheduleWithFixedDelay(
          {
-            // we block the thread used by the scheduler, as we execute inside the provided coroutine dispatcher
-            runBlocking {
-               launch(dispatcher) {
+            val previousCheck = jobs[name]
+            if(previousCheck == null || !previousCheck.isActive) {
+               jobs[name] = scope.launch(dispatcher) {
```

Comment:
if scope already has dispatcher we don't need to specify again right ?

Reply:
Great point! I pushed up a fix for that